@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -878,29 +868,16 @@ export function isPathLabelTooSmall(
     // Get the screen points that define the label's segments and create a path with
     // them.
     outScreenPoints.length = 0;
-    let anyPointVisible = false;
 
     for (const pt of textElement.points as THREE.Vector3[]) {
-        // Skip invisible points at the beginning of the path.
-        const screenPoint = anyPointVisible
-            ? screenProjector.project(pt, tmpScreenPosition)
-            : screenProjector.projectOnScreen(pt, tmpScreenPosition);
-        if (screenPoint === undefined) {
+        if (screenProjector.projectToScreen(pt, tmpScreenPosition) === undefined) {
             continue;
```

Comment:
By skipping invisible points in the middle of the path, you're altering the shape of the path. The text might not follow the path anymore.
Before we were only skipping invisible points at the beginning of the path, which is (more or less) ok. 
Probably we should still add the last point that was out of the screen, like in this case:


out&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;in
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
0--1--|---2
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;|    
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4--|---3
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;|
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5--|---6

Here, we were removing 0 and 1. We should probably keep 1 though. But surely we cannot skip 4 and 5 as you're doing with your change, otherwise you'll write on a straight line from 3 to 6.